source/common/http/http2/conn_pool.cc

```diff
@@ -132,6 +146,21 @@ void ConnPoolImpl::onConnectionEvent(ActiveClient& client, Network::ConnectionEv
     if (client.connect_timer_) {
       host_->cluster().stats().upstream_cx_connect_fail_.inc();
       host_->stats().cx_connect_fail_.inc();
+
+      // Raw connect failures should never happen under normal circumstances. If we have an upstream
```

Comment:
This bit is copied verbatim from the http1 conn pool. Should probably share that code somehow